adminSiteClient/ChartEditorPage.tsx

```diff
@@ -80,6 +82,15 @@ class TabBinder extends React.Component<{ editor: ChartEditor }> {
     }
 }
 
+function extractDetailsFromSyntax(str: string): [string, string][] {
+    const pattern = /\(hover::(\w+)::(\w+)\)/g
```

Comment:
I think we should elevate this regex and potential siblings to a shared place where it can be used by different pieces (e.g. in the admin we will want to check that the individual terms adhere to the same requirements - it would be bad if you can add numbers in terms in the admin but then when extracting them we don't find them because the regex excludes numbers).

This will come in handy I think when we add (include:: ...) as an alternative to hover::

I think we also want to be more permissive than \w, at least for the term. `int-$` would currently not work but might be more along what authors want to write than `international_dollars`. We should exclude : and () so that we don't get into parsing issues but I think adding a few more characters would be nice. In theory we could even allow whitespace but maybe that leads to more subtle issues than what it is worth. So maybe for now something like [A-Za-z0-9_\-$%\.+]? I'm not particularly fixed on any particular character and I think it makes sense to be more strict now and then extend but stuff like the $ sign is a pretty obvious candidate to be used IMHO.